pkg/shared/util/grpc_utils_test.go

```diff
@@ -0,0 +1,48 @@
+package util
+
+import (
+	"testing"
+
+	"github.com/numaproj/numaflow-go/pkg/info"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsCompatible(t *testing.T) {
+	tests := []struct {
+		name       string
+		serverInfo *info.ServerInfo
+		shouldErr  bool
+	}{
+		{
+			name: "Test with incompatible version",
+			serverInfo: &info.ServerInfo{
+				Protocol: info.UDS,
+				Language: info.Python,
+				Version:  "0.6.0",
+				Metadata: nil,
+			},
+			shouldErr: true,
+		},
+		{
+			name: "Test with compatible version",
```

Comment:
will this test fail if we update numaflow version or update version-mapping?

Reply:
yes so we would need to update this test each time, or we can just remove this test

Reply:
We don't want either. To make the code more unit test friendly, we can consider change the checkCompatibility method signature to something like `checkCompatibility(serverInfo, versionMapping, numaVersion)`, such that we can mock the input in unit tests.